File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let profit = 0;
    let min = prices[0];

    for(let i=1;i<prices.length;i++){
        if(min>prices[i])
        min=prices[i];

        profit = Math.max(profit, prices[i]-min);
    }

    return profit;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minimum_price = prices[0];
        int ans = 0;
        for(int i=1;i<prices.length;i++){
            int profit = prices[i] - minimum_price;
            if(profit>ans){
                ans = profit;
            } 

            if(minimum_price>prices[i])
            minimum_price = prices[i];
        }    
        return ans;
    }
}